VK/VK/FriendListScreen/FriendListTableViewController.swift

```diff
@@ -53,114 +46,90 @@ final class FriendListTableViewController: UITableViewController, UIViewControll
+        configScreen()
     }
 
-    private func getFirstLettersCount() {
-        sortedUsers = users.sorted(by: { user1, user2 in
-            guard
-                let user1FirstLetter = user1.name.first,
-                let user2FirstLetter = user2.name.first
-            else { return false }
-            return user1FirstLetter < user2FirstLetter
-        })
-
-        for user in sortedUsers {
-            for letter in lettersMap where letter.key == user.name.first {
-                sortedlettersMap[letter.key] = letter.value + 1
-                lettersMap[letter.key]? += 1
+    private func fetchFriends() {
+        NetworkServiceble.shared.fetchFriends(
+            parametersMap: [
+                Constants.fieldsName: Constants.fieldsValue,
+            ]
+        ) { friends in
+            let array = (friends as? ResponseWithFriends)?.response.items.map { friend in
+                NetworkUnit(
+                    name: "\(friend.firstName) \(friend.lastName)",
+                    description: friend.city?.title ?? String(Constants.emptyCharacter),
+                    avatarImageName: friend.photo ?? String(Constants.emptyCharacter),
+                    unitImageNames: [],
+                    id: friend.id
+                )
             }
+            self.makeFriendsSortedMap(friendsInfo: array ?? [])
+            self.tableView.reloadData()
         }
     }
 
-    private func calculateNumberOfSections() {
-        sortedlettersMap = lettersMap.filter { (_: Character, value: Int) in
-            guard value != 0 else { return false }
-            return true
-        }
+    private func configScreen() {
+        // configDotsLoader()
+        regCells()
+        configreTapHandler()
     }
 
-    private func sortLettersCountMap() {
-        lettersCountMap = sortedlettersMap.sorted(by: <)
+    private func setupCell() {
+        configScreen()
     }
 
-    private func putContactsInSections() {
-        var letterCounter = 0
-        var userCounter = 0
-        var array: [NetworkUnit] = []
-        for user in sortedUsers {
-            array.append(user)
-            userCounter += 1
-            if userCounter == lettersCountMap[safe: letterCounter]?.value {
-                userCounter = 0
-                letterCounter += 1
-                usersAtSections.append(array)
-                array.removeAll()
+    private func makeFriendsSortedMap(friendsInfo: [NetworkUnit]) {
+        var friendsMap: [Character: [NetworkUnit]] = [:]
+        for info in friendsInfo {
+            if let key = info.name.first {
+                if friendsMap[key] == nil {
+                    friendsMap[key] = [info]
+                } else {
+                    friendsMap[key]?.append(info)
+                    friendsMap[key]?.sort {
+                        $0.name.first ?? Constants.emptyCharacter > $1.name.first ?? Constants.emptyCharacter
+                    }
+                }
             }
         }
```

Comment:
```suggestion
        for info in friendsInfo {
            if let key = info.name.first {
                if friendsMap[key] == nil {
                    friendsMap[key] = [info]
                } else {
                    friendsMap[key]?.append(info)
                    friendsMap[key]?.sort {
                        $0.name.first ?? Constants.emptyCharacter > $1.name.first ?? Constants.emptyCharacter
                    }
                }
            }
        }
```
Большая вложенность, сложно читается. Я не разобрался. Наверняка можно сделать проще
понял зачем пустой character, но все же думаю можно упростить

Reply:
Поправил.